modules/localstack/localstack_test.go

```diff
@@ -165,3 +175,60 @@ func TestStartWithNetwork(t *testing.T) {
 	require.Equal(t, 1, len(networks))
 	require.Equal(t, "localstack-network", networks[0])
 }
+
+func TestStartV2WithNetwork(t *testing.T) {
+	ctx := context.Background()
+
+	nw, err := testcontainers.GenericNetwork(ctx, testcontainers.GenericNetworkRequest{
+		NetworkRequest: testcontainers.NetworkRequest{
+			Name: "localstack-network",
+		},
+	})
+	require.Nil(t, err)
+	assert.NotNil(t, nw)
+
+	localstack, err := StartContainer(
+		ctx,
+		OverrideContainerRequest(testcontainers.ContainerRequest{
+			Image:          "localstack/localstack:2.0.0",
+			Networks:       []string{"localstack-network"},
+			NetworkAliases: map[string][]string{"localstack-network": {"localstack"}},
+		}),
+	)
+	require.Nil(t, err)
+	assert.NotNil(t, localstack)
+
+	cli, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image:      "amazon/aws-cli:2.7.27",
+			Networks:   []string{"localstack-network"},
+			Entrypoint: []string{"tail"},
+			Cmd:        []string{"-f", "/dev/null"},
+			Env: map[string]string{
+				"AWS_ACCESS_KEY_ID":     "accesskey",
+				"AWS_SECRET_ACCESS_KEY": "secretkey",
+				"AWS_REGION":            "eu-west-1",
+			},
+		},
+		Started: true,
+	})
+	require.Nil(t, err)
+	assert.NotNil(t, cli)
+
+	c, output, err := cli.Exec(ctx, []string{"/usr/local/bin/aws", "sqs", "create-queue", "--queue-name", "baz", "--region", "eu-west-1",
+		"--endpoint-url", "http://localstack:4566", "--no-verify-ssl"})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if c != 0 {
+		t.Fatalf("AWS CLI command was executed, expected return code 1, got %v", c)
```

Comment:
The tests are currently failing here. I'm debugging them locally atm

Reply:
Double checked that it fails locally. Will try to understand why and propose a fix

Reply:
This is interesting! the same config works in tc-java. I was wondering if the network name was making those tests flaky but look like is not. So, if I understood correctly the suggestion is to add the wait strategies as part of the container request, right? Although, the check is against the cli output. I think we can store the result in a file and check it later but doesn't seem to simplify the pr

Reply:
I'd still create the CLI container, but instead of directly using the Exec method of the container, move it to a wait.ForExec(...).WithResponseMatcher(). If the container is not created without error, then the tests should fail

Does it sound good to you?

Reply:
I think we have the same issue 🤔

Reply:
🤔 Could it be caused by the startup timeout not being enough depending on the machine?

I've seen some issues regarding that in the couchbase module in a non-deterministic manner. Restarting its CI build fixes it. Could it be the case for this?